projects/canopy/src/lib/utils/dynamic-style.service.ts

```diff
@@ -0,0 +1,97 @@
+import { Injectable } from '@angular/core';
+
+import { BreakpointValues } from '../shared/breakpoints.interface';
+
+interface Rule {
+  selector: string;
+  declaration: string;
+}
+interface RuleRepsonive {
+  selector: string;
+  declaration: string;
+  breakpoint: BreakpointValues;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+/**
+ * Dynamically add CSS rules to a <style> tag in the <head>
+ */
+export class DynamicStyleService {
+  styleTag = null;
+  selectors = [];
+  mediaQueries = {};
+
+  private addStyleTag(): void {
+    this.styleTag = document.createElement('style');
+    this.styleTag.type = 'text/css';
+    document.getElementsByTagName('head')[0].appendChild(this.styleTag);
+  }
+
+  private createMediaQuery(breakpoint: BreakpointValues): string {
+    if (this.mediaQueries[breakpoint]) {
+      return;
+    }
+    this.mediaQueries[breakpoint] = [];
+    return `@media(min-width:${breakpoint}){}`;
+  }
+
+  addRules(rules: Array<Rule>): void {
+    if (!this.styleTag) {
```

Comment:
quite minor, but the assumption being that if this service is loaded then it will be used, does that mean we could do this in the `constructor`?

Reply:
Ah good point, I agree - also cuts a little duplication. Moved it ✅